### DISCUSSION TITLE
What's the "save" button in upper right corner of a code snippet ?

### DISCUSSION THREAD

**Discussion by 1c383r9** [open-webui/open-webui#6576] [Q&A] [answered]
I thought it would download the snippet but nothing happened on my self-host site.

What's the original purpose of this button?

Appreciate it!!

e.g.(Mine is in Mandarin. The other button is "copy")
![image](https://github.com/user-attachments/assets/33feb068-4b12-4b5a-a9f5-ec38e0261acc)

**Comment by peuportier** [comment, marked as answer]
If you are generated some code like let said you make something in html , then it shows the result on Artifact. You can directly to the code and modify it. Then press save to see the obtain result.

**Reply by 1c383r9** [reply]
Thx!!!

**Reply by peuportier** [reply]
welcome , enjoy OWUI

**Reply by AlterFalter** [reply]
Thank you for the explanation. Could this possibly be renamed to "Update Preview" or alternatively have a tooltip that explains the button to the user? Many users from us expected this to be a "download as a file" feature

**Reply by AlterFalter** [reply]
Third solution would be to only show it when the Preview is actually visible and hide it otherwise, since it does nothing anyway in that cases.
